Dockerfile

```diff
@@ -1,19 +1,91 @@
-FROM node:14-alpine
+# ==
+# builder
+# ==
+
+# application builder
+FROM node:16 AS builder
 
-# directory for the app in the container
 WORKDIR /usr/app
 
-# copies all the app's files from host into the container folder which
-# might include the node_modules dir if npm install executed in the host
+# copy package.json and package-lock.json files
+COPY package*.json .
+
+# install dependencies including local development tools
+RUN npm install
+
+# copy the rest of the content
 COPY . /usr/app
 
-# removes any existing node_modules folder
-# this prevents the host's node_modules from being used in the container
-# which could cause issues with native binaries such as node_sass.
-RUN rm -rf /usr/app/node_modules/
+# disable telemetry when building the app
+ENV NUXT_TELEMETRY_DISABLED=1
+
+# build the application and generate a distribution package
+RUN npm run build
+
+# ==
+# development
+# ==
+# application for development purposes
+FROM node:16 AS dev
+
+WORKDIR /usr/app
 
+ENV NODE_ENV=development
 ENV CYPRESS_INSTALL_BINARY=0
 
-RUN npm install -g npm@7.21.0 && \
-    npm install && \
-    npm run i18n:get-translations
+# copy files from local machine
+COPY . /usr/app
+
+# disable telemetry when building the app
+ENV NUXT_TELEMETRY_DISABLED=1
+
+# install dependencies (development dependencies included)
+RUN npm install
+
+# expose port 8443
+EXPOSE 8443
+
+# run the application in development mode
+ENTRYPOINT [ "npm", "run", "dev" ]
+
+# ==
+# production
+# ==
+# application package (for production)
+FROM node:alpine AS app
+
+WORKDIR /usr/app
+
+ENV NODE_ENV=production
```

Comment:
```suggestion

```

We'll actually want to remove this, related to #450. We want to be able to use NODE_ENV as `staging` or as `production`. Can this be moved to the actions, maybe @rbadillap ? By the way, Nuxt builds the app the same in staging or prod mode, so you don't have to worry about differences in the built files based on this key.

Reply:
based on internal conversations, this should be kept as it